src/main/java/uk/gov/hmcts/reform/sscs/messaging/HmcHearingsEventTopicListener.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.sscs.messaging;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusException;
+import com.azure.messaging.servicebus.ServiceBusFailureReason;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Data
+@SuppressWarnings({
+    "PMD.LawOfDemeter",
+    "PMD.CompareObjectsWithEquals",
+    "PMD.CloseResource"
+})
+public final class HmcHearingsEventTopicListener {
+
+    private final HmcHearingsTopicListenerProperties properties;
+
+    @SuppressWarnings("unused")
+    public void receiveMessages() {
+        CountDownLatch countdownLatch = new CountDownLatch(1);
+
+        ServiceBusProcessorClient processorClient = new ServiceBusClientBuilder()
+            .connectionString(properties.getConnectionString())
+            .processor()
+            .topicName(properties.getTopicName())
+            .subscriptionName(properties.getSubName())
+            .processMessage(HmcHearingsEventTopicListener::processMessage)
+            .processError(context -> processError(context, countdownLatch))
+            .buildProcessorClient();
+
+        processorClient.start();
+        try {
+            TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException e) {
+            log.error("Client processing interrupted: {}%n ", e.getMessage());
+        }
+        processorClient.close();
+    }
+
+    @SuppressWarnings({"PMD.EmptyIfStmt", "checkstyle:EmptyBlock"})
+    private static void processMessage(ServiceBusReceivedMessageContext context) {
+        ServiceBusReceivedMessage message = context.getMessage();
+        HmcMessage hmcMessage = message.getBody().toObject(HmcMessage.class);
+        String hmctsServiceID = hmcMessage.getHmctsServiceID();
+
+        if (hmctsServiceID.contains("BBA3")) {
+            //TODO process all messages with BBA3
+        }
+
+        log.info("Processing message. Session: {}, Sequence #: {}. Contents: {}%n", message.getMessageId(),
+                 message.getSequenceNumber(), message.getBody());
+    }
+
+    private void processError(ServiceBusErrorContext context, CountDownLatch countdownLatch) {
+        log.error("Error when receiving messages from namespace: '{}'. Entity: '{}'%n",
```

Comment:
Have you discussed this method with Thomas? His ticket concerns the errors that are returned, the two tickets should link together here.